### PR TITLE
make default branch configurable via env var; don't replace + with - …

### PR DESCRIPTION
**Pull request by mlilien**
…for all gitversion variables, only for variables with suffix DockerLabel

### CODE REVIEW

**Comment by siredmar** [comment]
Looks good to me! Good adition to bb-gitversion. 
Any idea when this gets released?

**Comment by mlilien** [comment]
> Looks good to me! Good adition to bb-gitversion.
> Any idea when this gets released?

Should be within the next 2 weeks.

File: gitversion/entrypoint.sh [hunk 1]
```diff
@@ -27,25 +27,31 @@ fi
...
 else
     targetVersion="InformationalVersion"
 fi
 targetVersionInfo=$(cat ${JSON_DIR}/gitversion.json | jq -r .${targetVersion})
-cat ${JSON_DIR}/gitversion.json | jq --arg value ${targetVersionInfo} '.+{BranchVersion: $value }' > ${JSON_DIR}/gitversion_append.json
-mv ${JSON_DIR}/gitversion_append.json ${JSON_DIR}/gitversion.json
+cat ${JSON_DIR}/gitversion.json | jq --arg value ${targetVersionInfo} '.+{BranchVersion: $value }' > /tmp/gitversion2.json
+targetVersionInfoDockerLabel=$(sed 's/+/-/g' <<< ${targetVersionInfo})
+cat /tmp/gitversion2.json | jq --arg value ${targetVersionInfoDockerLabel} '.+{BranchVersionDockerLabel: $value }' > /tmp/gitversion3.json
+#fullSemver=$(cat ${JSON_DIR}/gitversion.json | jq -r .FullSemVer)
```
Comment (JoergZeidler) [inline gitversion/entrypoint.sh]: could be removed?

**Review by JoergZeidler — COMMENTED** [review]
could we not generate BranchVersion as it was, all building-blocks has to be adapted to new env instead.

**Comment by mlilien** [comment]
> could we not generate BranchVersion as it was, all building-blocks has to be adapted to new env instead.

Makes no sense to me, the doku states we are FullSemVer compatible. So I marked the exceptions explicitly.
You don't need to upgrade all buildingblocks, but if you do, you have to adapt accordingly to the Changelog/Readme remarks.

File: gitversion/entrypoint.sh (hunk 1 above)
Comment (mlilien) [reply gitversion/entrypoint.sh]: done

**Review by JoergZeidler — APPROVED** [review]

**Comment by siredmar** [comment]
> > could we not generate BranchVersion as it was, all building-blocks has to be adapted to new env instead.
> 
> Makes no sense to me, the doku states we are FullSemVer compatible. So I marked the exceptions explicitly.
> You don't need to upgrade all buildingblocks, but if you do, you have to adapt accordingly to the Changelog/Readme remarks.

This comment makes absolute sense to me. Building blocks can use any version of bb-gitversion they want.

**Pull request merged**